src/back/space/src/api/invitation.cpp

```diff
@@ -0,0 +1,164 @@
+#include "invitation.hpp"
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+#include "../../../shared/errors.hpp"
+#include "../../../shared/paging.hpp"
+#include "../../../shared/parse/request.hpp"
+#include "../model/role.hpp"
+#include "../model/invitation_serialize.hpp"
+
+namespace svetit::space::handlers {
+
+Invitation::Invitation(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Invitation::HandleRequestJsonThrow(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body,
+	server::request::RequestContext&) const
+{
+	formats::json::ValueBuilder res;
+
+	const auto userId = req.GetHeader(headers::kUserId);
+	if (userId.empty())
+		throw errors::Unauthorized{};
+
+	try {
+		switch (req.GetMethod()) {
+		case server::http::HttpMethod::kGet:
+			return GetList(req, res, userId);
+		case server::http::HttpMethod::kPost:
+			return Post(req, body, res, userId);
+		case server::http::HttpMethod::kPut:
+			return ChangeRole(req, body, res, userId);
+		case server::http::HttpMethod::kPatch:
+			return Join(req, res, userId);
+		case server::http::HttpMethod::kDelete:
+			return Delete(req, res, userId);
+		default:
+			throw std::runtime_error("Unsupported");
+			break;
+		}
+	} catch(const errors::Unauthorized& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kUnauthorized);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::BadRequest& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::NotFound& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kNotFound);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::NotModified& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kNotModified);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::Forbidden403& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kForbidden);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const std::exception& e) {
+		LOG_WARNING() << "Fail to process invitation handle with method: "
+			<< req.GetMethodStr() << " err: " << e.what();
+		res["err"] = "Fail to process invitation";
+		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+	}
```

Comment:
Добавь пожалуйста файл shared/errors_catchit.сpp/hpp с примерно таким содержимым в сpp:

```cpp
namespace svetit::errors {

formats::json::Value CatchIt(const server::http::HttpRequest& req)
{
	static constexpr const char* errKey = "err";
	formats::json::ValueBuilder res;

	auto eptr = std::current_exception();
	try {
		if (eptr)
			std::rethrow_exception(eptr);
	} catch(const Exception& e) {
		req.SetResponseStatus(e.GetHttpStatus());
		res[errKey] = e.what();
	} catch(const std::exception& e) {
		LOG_ERROR()
			<< "Method: " << req.GetMethodStr()
			<< " Url: " << req.GetUrl()
			<< " Err: " << e.what()
			<< " Body: " << req.RequestBody().substr(0, 1024);
		res[errKey] = "Internal error";
		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
	} catch (...) {
		throw;
	}
	return res.ExtractValue();
}

} // namespace svetit::errors
```

И потом везде можно использовать это вот так:
```suggestion
	} catch(...) {
		return errors::CatchIt(req);
	}
```

Reply:
Готово.